synapse/util/manhole.py

```diff
@@ -105,7 +105,8 @@ def manhole(settings: ManholeConfig, globals: Dict[str, Any]) -> Factory:
     factory.privateKeys[b"ssh-rsa"] = priv_key  # type: ignore[assignment]
     factory.publicKeys[b"ssh-rsa"] = pub_key  # type: ignore[assignment]
 
-    return factory
+    # ConchFactory is a Factory, not a ServerFactory, but they are identical.
+    return factory  # type: ignore[return-value]
```

Comment:
Maybe this should be a `cast`?

Reply:
As above, mypy won't do any analysis based on the type of `factory` because it's the last line of this function. I think this is fine.